binding/Binding/SKDrawable.cs

```diff
@@ -0,0 +1,164 @@
+
+			SkiaApi.sk_manageddrawable_set_delegates (
+				Marshal.GetFunctionPointerForDelegate (fDraw),
+				Marshal.GetFunctionPointerForDelegate (fGetBounds),
+				Marshal.GetFunctionPointerForDelegate (fNewPictureSnapshot));
+		}
+
+		protected SKDrawable ()
+			: this (true)
+		{
+		}
+
+		protected SKDrawable (bool owns)
+			: base (SkiaApi.sk_manageddrawable_new (), owns)
+		{
+			if (Handle == IntPtr.Zero) {
+				throw new InvalidOperationException ("Unable to create a new SKAbstractDrawable instance.");
+			}
+
+			managedDrawables.TryAdd (Handle, this);
+		}
+
+		[Preserve]
+		internal SKDrawable (IntPtr x, bool owns)
+			: base (x, owns)
+		{
+		}
+
+		private void DisposeFromNative ()
+		{
+			Interlocked.Exchange (ref fromNative, 1);
+			Dispose ();
+		}
+
+		protected override void Dispose (bool disposing)
+		{
+			if (disposing) {
+				managedDrawables.TryRemove (Handle, out var managedDrawable);
+			}
+
+			if (Interlocked.CompareExchange (ref fromNative, 0, 0) == 0 && Handle != IntPtr.Zero && OwnsHandle) {
+				SkiaApi.sk_manageddrawable_destroy (Handle);
+			}
+
+			base.Dispose (disposing);
+		}
+
+		public uint GenerationID => SkiaApi.sk_drawable_get_generation_id (Handle);
+
+		public SKRect Bounds {
+			get {
+				SkiaApi.sk_drawable_get_bounds (Handle, out var bounds);
+				return bounds;
+			}
+		}
+
+		public void Draw (SKCanvas canvas, ref SKMatrix matrix)
+		{
+			SkiaApi.sk_drawable_draw (Handle, canvas.Handle, ref matrix);
+		}
+
+		public void Draw (SKCanvas canvas, float x, float y)
+		{
+			var matrix = SKMatrix.MakeTranslation (x, y);
+			SkiaApi.sk_drawable_draw (Handle, canvas.Handle, ref matrix);
+		}
+
+		public SKPicture NewPictureSnapshot ()
+		{
+			return GetObject<SKPicture> (SkiaApi.sk_drawable_new_picture_snapshot (Handle));
+		}
+
+		public void NotifyDrawingChanged ()
+		{
+			SkiaApi.sk_drawable_notify_drawing_changed (Handle);
+		}
+
+		protected virtual void OnDraw (SKCanvas canvas)
+		{
+		}
+
+		protected virtual SKRect OnGetBounds ()
+		{
+			return new SKRect ();
+		}
+
+		protected virtual SKPicture OnNewPictureSnapshot ()
+		{
+			var recorder = new SKPictureRecorder ();
+			var canvas = recorder.BeginRecording (Bounds);
+			Draw (canvas, 0, 0);
+			return recorder.EndRecording ();
+		}
+
+		// unmanaged <-> managed methods (static for iOS)
+
+		[MonoPInvokeCallback (typeof (draw_delegate))]
+		private static void DrawInternal (IntPtr managedDrawablePtr, IntPtr canvas)
+		{
+			AsManagedDrawable (managedDrawablePtr).OnDraw (GetObject<SKCanvas> (canvas));
+		}
+
+		[MonoPInvokeCallback(typeof(getBounds_delegate))]
+		private static void GetBoundsInternal (IntPtr managedDrawablePtr, out SKRect rect)
+		{
+			rect = AsManagedDrawable (managedDrawablePtr).OnGetBounds ();
+		}
+
+		[MonoPInvokeCallback(typeof(newPictureSnapshot_delegate))]
+		private static SKPicture NewPictureSnapshotInternal (IntPtr managedDrawablePtr)
```

Comment:
This has to be a blittable type - `IntPtr` for the handle:

```suggestion
		private static IntPtr NewPictureSnapshotInternal (IntPtr managedDrawablePtr)
```